src/GraphQL.Tests/Utilities/SchemaVisitorTests.cs

```diff
@@ -91,5 +94,234 @@ public override void VisitField(FieldType field)
                 });
             }
         }
+
+        [Fact]
+        public void can_apply_custom_directive_when_graph_type_first()
+        {
+            var objectType = new ObjectGraphType();
+            objectType.Field<StringGraphType>()
+                .Name("hello")
+                .Resolve(_ => "Hello World!")
+                .Directive("upper", new UppercaseDirectiveVisitor());
+
+            var directivesMetadata = objectType.Fields.First().GetDirectives();
+            directivesMetadata.ShouldNotBeNull();
+            directivesMetadata.Count.ShouldBe(1, "Only 1 directive should be added");
+            directivesMetadata.ContainsKey("upper").ShouldBeTrue();
+
+            var queryResult = CreateQueryResult("{ 'hello': 'HELLO WORLD!' }");
+            var schema = new Schema { Query = objectType };
+
+            AssertQuery(_ =>
+            {
+                _.Schema = schema;
+                _.Query = "{ hello }";
+            }, queryResult);
+        }
+
+        public class RegisterTypeDirectiveVisitor : SchemaDirectiveVisitor
```

Comment:
Consider to move visitors to separate files.

Reply:
Moved